vip_hci/hci_postproc.py

```diff
@@ -16,18 +16,50 @@
 from .metrics import snrmap_fast, snrmap
 from .pca import pca
 import pickle
+import numpy as np
 
 
 class HCIPostProcAlgo(BaseEstimator):
     """ Base HCI post-processing algorithm class.
     """
+    THRESHOLDS_5 = [0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5]
+    THRESHOLDS_1 = np.linspace(0.1, 0.99, 10).tolist()
     def print_parameters(self):
         """ Printing out the parameters of the algorithm.
         """
         dicpar = self.get_params()
         for key in dicpar.keys():
             print("{}: {}".format(key, dicpar[key]))
 
+    def store_args(self, kwargs, *skip):
+        # TODO: this could be integrated with sklearn's BaseEstimator methods
+        for k in kwargs:
+            if k == "self" or k in skip:
+                continue
+            setattr(self, k, kwargs[k])
+
+
+    def _get_dataset(self, dataset=None, verbose=True):
+        if dataset is None:
+            dataset = self.dataset
+            if self.dataset is None:
+                raise ValueError("no dataset specified!")
+        else:
+            self.dataset = dataset # needed for snr map generation
+            if verbose:
+                #print("self.dataset overwritten with the one you provided.")
```

Comment:
Why is the print commented?

Reply:
I am not sure if it should be kept or not. There are two ways of providing a dataset

1. through the constructor, that would be the "normal use"
2. through the `run` method, which is used when the same algo with the same parameters should be run on multiple datasets (e.g. ROC curve generation)

When a user supplies a dataset on *both* places, that is where the `print` would output something. I can think of two scenarios where this could happen:

1. The user first initializes a `HCIPostProcAlgo`, tests it on one single injection (by passing a dataset to the constructor), tunes the parameters, and then decides to build a ROC curve. In that case, the `print` statement is useless, as by design the `EvalRoc` overwrites the original dataset by the one with the injections.
2. The user just wants to run an `HCIPostProcAlgo` on one dataset. By mistake he passes a dataset to the constructor *and* to the `run` method. In that case, it would be useful to warn the user that the dataset passed to the constructor is ignored and overwritten.

Reply:
Let's leave this question open for the now